### PR TITLE
feat(route): Add bloomberg rss

### LINKED ISSUE #10936: 彭博社 Bloomberg
### 类型

金融

### 网站地址

https://www.bloomberg.com/

### 网站描述

全球顶级金融终端，全球金融资讯平台，全球金融终端领导者

### 需要生成什么内容？

----------------------------------------（可选地区）
首页头条：https://www.bloomberg.com/ asia | us | europe | uk | middleeast | africa
Markets：https://www.bloomberg.com/markets
Politics：https://www.bloomberg.com/politics
industries：https://www.bloomberg.com/industries
technology：https://www.bloomberg.com/technology
............支持所有栏目
image

### 额外描述

_No response_

### 这不是重复的 RSS 请求

- [X] 我已经搜索了[现有 issue](https://github.com/DIYgod/RSSHub/issues) 和 [pull requests](https://github.com/DIYgod/RSSHub/pulls)，以确保该 RSS 尚未被请求。

### PR DESCRIPTION
**Pull request by bigfei**
## 该 PR 相关 Issue / Involved issue

Close #10936

## 完整路由地址 / Example for the proposed route(s)

```routes
/bloomberg
/bloomberg/bpol
/bloomberg/bbiz
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [ ] 中文文档 CN
  - [x] 英文文档 EN
- [x] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [x] 反爬/频率限制 anti-bot or rate limit?
  - [x] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note
Bloomberg has anti-spider system deployed, so it is better to run on self deployment.

### CODE REVIEW

**Comment by bigfei** [comment]
@TonyRL  Can you try to deploy it standalone and test it locally?

File: lib/v2/bloomberg/utils.js [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+
+    return urls.slice(0, ctx.query.limit ? parseInt(ctx.query.limit) : 50).map((u) => {
+        const news = u['news:news'][0];
+        const item = {
+            link: u.loc[0],
+            title: news['news:title'][0],
+            pubDate: parseDate(news['news:publication_date'][0]),
+        };
+        return item;
+    });
```
Comment (TonyRL) [inline lib/v2/bloomberg/utils.js:69]: You can use `cheerio` to handle xml as well
```suggestion
    const $ = cheerio.load(resp.data, {
        xml: {
            xmlMode: true,
        },
    });
    const urls = $('urlset url');
    return urls.toArray().slice(0, ctx.query.limit ? parseInt(ctx.query.limit) : 50).map((u) => {
        u = $(u);
        const item = {
            title: u.find('news\\:title').text(),
            link: u.find('loc').text(),
            pubDate: parseDate(u.find('news\\:publication_date').text()),
        };
        return item;
    });
```

File: lib/v2/bloomberg/utils.js [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+                        title: article_json.headline || item.title,
+                        link: article_json.canonical || item.link,
+                        guid: `bloomberg:${article_json.id}`,
+                        description: (await processBody(article_json.body, article_json)).replaceAll(emptyRegex, ''),
+                        pubDate: item.pubDate,
+                        author: article_json.authors?.map((a) => a.name).join(', ') ?? [],
+                    };
+                    return rss_item;
+                } else {
+                    const article_json = JSON.parse(
```
Comment (TonyRL) [inline lib/v2/bloomberg/utils.js:145]: You can check if `res.redirectUrls` contains `https://www.bloomberg.com/tosv2.html...` here and inside `processVideo` method

File: lib/v2/bloomberg/utils.js (hunk 1 above)
Comment (bigfei) [reply lib/v2/bloomberg/utils.js]: looking good for native async support

File: lib/v2/bloomberg/utils.js [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+
+                    caption: episode.description || ($(e).find('[class$="text"]').html()?.trim() ?? ''),
+                    credit: (episode.credits.map((c) => c.name).join(', ') ?? []) || ($(e).find('[class$="credit"]').html()?.trim() ?? ''),
+                };
+            }
+            new_figure = art(path.join(__dirname, 'templates/audio_media.art'), audio);
+        } else if (imageType === 'video') {
+            if (story_json.videoAttachments) {
+                const attachment = story_json.videoAttachments[$(e).data('id')];
+                // eslint-disable-next-line no-await-in-loop
```
Comment (TonyRL) [inline lib/v2/bloomberg/utils.js:273]: You can use [for await...of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of) loop

**Comment by bigfei** [comment]
@TonyRL  Can you have a review for my pull request finally?

**Comment by Majdian** [comment]
Hi

**Comment by Majdian** [comment]
How I'm can mike many?

**Comment by CarlaRobinson12** [comment]
_[[](https://github.com/trimstray/the-book-of-secret-knowledge/commit/62f533707bbfabe2c71aa6c659802a917684cc0a)](https://github.com/trimstray/the-book-of-secret-knowledge/commit/62f533707bbfabe2c71aa6c659802a917684cc0a)_

**Pull request merged**